app/mailers/user_mailer.rb

```diff
@@ -269,10 +269,11 @@ def related_work_notification(user_id, related_work_id)
   end
 
   # Emails a recipient to say that a gift has been posted for them
-  def recipient_notification(user_id, work_id, collection_id=nil)
+  def recipient_notification(user_id, work_id, collection_id = nil)
     @user = User.find(user_id)
     @work = Work.find(work_id)
-    @collection = Collection.find(collection_id) if collection_id
+    # If we've supplied a collection_id, make sure the collection mods and work creators have approved the work's inclusion in the collection before adding the collection name to the email
+    @collection = Collection.find(collection_id) if collection_id && @work.collection_items.where(collection_id: collection_id, collection_approval_status: CollectionItem::APPROVED, user_approval_status: CollectionItem::APPROVED).present?
```

Comment:
> there is a minor performance issue with the definition of `approved_collections`

Do we have a ticket for that yet?

Reply:
No. I originally ran into the issue when I was working on the n+1 work code (since it goes from performance issue to actual bug when you try to call `Work.includes(:approved_collections)`), so I included a fix for it in the PR that I submitted for that, but then I had to close the PR because it was incompatible with the search upgrade.

It's probably worth a separate ticket, because my WIP n+1 work code has become a lot more complex since then, and the fix for `approved_collections` is only a few lines. Easier to merge in smaller increments.

Reply:
[AO3-5594](https://otwarchive.atlassian.net/browse/AO3-5594) is all yours.

Reply:
Dang I didn't see AO3-5593, I'll close mine as duplicate.

Reply:
No worries! Yours had a better write-up.